src/DynamoCoreWpf/ViewModels/Search/NodeAutoCompleteSearchViewModel.cs

```diff
@@ -799,45 +799,31 @@ internal void AddCluster(ClusterResultItem ClusterResultItem)
                     index++;
                 }
             }
-
-            
-            clusterConnections.ForEach(connection =>
+            using (var undoGroup = wsViewModel.Model.UndoRecorder.BeginActionGroup())
             {
-                // Connect the nodes
-                var sourceNode = clusterMapping[connection.StartNode.NodeId];
-                var targetNode = clusterMapping[connection.EndNode.NodeId];
-                // The port index is 1- based (currently a hack and not expected from service)
-                var sourcePort = sourceNode.OutPorts.FirstOrDefault(p => p.PortModel.Index == connection.StartNode.PortIndex - 1);
-                var targetPort = targetNode.InPorts.FirstOrDefault(p => p.PortModel.Index == connection.EndNode.PortIndex - 1);
-                var commands = new List<DynamoModel.ModelBasedRecordableCommand>
-                    {
-                        new DynamoModel.MakeConnectionCommand(sourceNode.Id.ToString(), connection.StartNode.PortIndex - 1, PortType.Output, DynamoModel.MakeConnectionCommand.Mode.Begin),
-                        new DynamoModel.MakeConnectionCommand(targetNode.Id.ToString(), connection.EndNode.PortIndex - 1, PortType.Input, DynamoModel.MakeConnectionCommand.Mode.End),
-                    };
-                commands.ForEach(c =>
+                clusterConnections.ForEach(connection =>
                 {
-                    try
+                    // Connect the nodes    
+                    var sourceNode = clusterMapping[connection.StartNode.NodeId];
+                    var targetNode = clusterMapping[connection.EndNode.NodeId];
+                    // The port index is 1- based (currently a hack and not expected from service)
+                    var sourcePort = sourceNode.OutPorts.FirstOrDefault(p => p.Index == connection.StartNode.PortIndex - 1);
+                    var targetPort = targetNode.InPorts.FirstOrDefault(p => p.Index == connection.EndNode.PortIndex - 1);
+
+                    var connector = ConnectorModel.Make(sourceNode, targetNode, connection.StartNode.PortIndex - 1, connection.EndNode.PortIndex - 1);
```

Comment:
I had to change this to the following to run in debug locally:
```
   var sourcePort = sourceNode.OutPorts.FirstOrDefault(p => p.PortModel.Index == connection.StartNode.PortIndex - 1);
   var targetPort = targetNode.InPorts.FirstOrDefault(p => p.PortModel.Index == connection.EndNode.PortIndex - 1);

   var connector = ConnectorModel.Make(sourceNode.NodeModel, targetNode.NodeModel, connection.StartNode.PortIndex - 1, connection.EndNode.PortIndex - 1);
```

Reply:
That's strange, the `OutPorts` property should already be a collection of PortModel so shouldn't need to get PortModel again
![image](https://github.com/user-attachments/assets/b9d11923-df3b-4e57-ad76-9cdf5733e934)

Reply:
hmm, I wonder why I am seeing the collection be NodeViewModel for some reason: https://github.com/DynamoDS/Dynamo/blob/af3256839516845f080abdba29bc22bc13139963/src/DynamoCoreWpf/ViewModels/Search/NodeAutoCompleteSearchViewModel.cs#L773

Reply:
yeah, it's a dictionary from there, then we get individual node here: https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCoreWpf/ViewModels/Search/NodeAutoCompleteSearchViewModel.cs#L807
Then we get the ports and compare the index with response to determine which port to connect

Reply:
Yeah, that appears to be returning a NodeViewModel is what I am saying.

Reply:
Oh, maybe the automatic merge conflicts resolve failed, I will check, thanks!

Reply:
I think so yes, in my local integration for these changes on top of master it looks good.

Reply:
This merge conflicts should be addressed now  @johnpierson @BogdanZavu